src/autodrop.hpp

```diff
@@ -0,0 +1,32 @@
+template<class T>
```

Comment:
While the new C++11 smart pointers allow specifying a Deleter template parameter, ``auto_ptr`` does not. Since I thought I had to stick to ``auto_ptr``, I decided the easiest way was to write an own smart pointer class for pointers that have to be ``drop()``ped instead of ``delete``d.